consumer/consumererror/consumererror.go

```diff
@@ -0,0 +1,61 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package consumererror // import "go.opentelemetry.io/collector/consumer/consumererror"
+
+import (
+	"errors"
+	"fmt"
+
+	"google.golang.org/grpc/status"
+)
+
+func New(errs ...error) error {
+	return errors.Join(errs...)
+}
+
+type StatusError struct {
+	error
+	httpStatus *int
+	grpcStatus *status.Status
+}
+
+func (se *StatusError) Error() string {
+	if se.httpStatus != nil {
+		return fmt.Sprintf("HTTP Status %d", *se.httpStatus)
+	} else {
+		return fmt.Sprintf("gRPC Status %s", se.grpcStatus.Code().String())
+	}
+}
+
+func (se *StatusError) HTTPStatus() int {
```

Comment:
Yeah, that would make sense. We could also make it return `*int` to indicate "no status set" if you think that would be better.

Reply:
I've updated this such that if the user creates an error with a `nil` pointer to a grPC `status.Status` object, we return `false` for the second return value. Thinking about this again, I think I can see a few other possible routes for this:

1. Should we just default to grpc `codes.Internal` if the user does this? To me this seems like an issue with the exporter that created the error (why create a gRPC status error if there's no code to report?) that the upstream component really won't have the context to make a more intelligible decision about.
2. Should we keep `(status int, ok bool)` and not do gRPC->HTTP conversion? I wanted to keep this as the default because I believe this will generally be used by receivers who want to return a code for a particular transport and won't care as much about which transport was used during exporting.

Reply:
Option 1 seems reasonable to me, this is an invalid situation and there is not much that the upstream component can do, so I am in favor of going with it

Reply:
Updated. The codes also match the defaults in our conversions, so I think this is probably the best way forward.